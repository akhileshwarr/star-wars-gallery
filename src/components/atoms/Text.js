import React, { useContext } from 'react';
import { ThemeContext } from '../../context/theme';
import PropTypes from 'prop-types';

export const Text = ({children }) => {

    const theme = useContext(ThemeContext);

    return (
        <span className = "genralSpan" style = {{color:theme.foreground}}>{children}</span>
    );
  };

  Text.propTypes = {
    children:  PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array
    ])
  };
  
  Text.defaultProps = {
    children: null
  };