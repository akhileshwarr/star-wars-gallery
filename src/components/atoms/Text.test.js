import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from './Text.js';

test('/Text renders snapshot', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });
