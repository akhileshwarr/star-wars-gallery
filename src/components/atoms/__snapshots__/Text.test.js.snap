// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Text renders snapshot 1`] = `
<span
  className="genralSpan"
  style={
    Object {
      "color": "#ffffff",
    }
  }
/>
`;
